apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/trace/TraceSegment.java

```diff
@@ -89,6 +89,13 @@ public void ref(TraceSegmentRef refSegment) {
         if (refs == null) {
             refs = new LinkedList<>();
         }
+        /*
+         * There maybe an exception in some plugins, which causes the span to fail to exit,
+         * which results in the ThreadLocal not being cleaned up, and this data is increasing.
+         */
+        if (refs.size() >= Config.Agent.TRACE_SEGMENT_REF_LIMIT_PER_SEGMENT) {
```

Comment:
Similar to https://github.com/apache/skywalking/pull/6715/files#r610564498, check the code, please.